packages/integration/src/gerrit/core.ts

```diff
@@ -156,14 +157,18 @@ export function getAuthenticationPrefix(
 export function getGitilesAuthenticationUrl(
   config: GerritIntegrationConfig,
 ): string {
-  const parsedUrl = new URL(config.gitilesBaseUrl!);
-  parsedUrl.pathname = parsedUrl.pathname.replace(/\/?$/, '');
-  parsedUrl.pathname = parsedUrl.pathname.replace(
-    /\/([^\/]*)$/,
-    `${getAuthenticationPrefix(config)}$1`,
-  );
-
-  return parsedUrl.toString();
+  if (config.gitilesBaseUrl!.startsWith(config.baseUrl!)) {
```

Comment:
I'm not seeing any obvious guarantee that these are actually set. It'd likely be best to return `undefined` in case the necessary fields aren't set, or otherwise throw errors. Let's avoid any `!` in this code

Reply:
Defaults are setup in the config-reader: https://github.com/GLundh/backstage/blob/2d2fc9d20ebb07cd53e47e36bbe2b428209b7b99/packages/integration/src/gerrit/config.ts#L73

But of course I agree. I'll fix it.